### PR TITLE
flatpak support for Unknown Horizons

### PR DESCRIPTION
**Pull request by tintou**
### Please confirm your submission meets all the criteria

- [X] I have read the [App Requirements][reqs] and [App Maintenance][maint] pages.
- [X] My pull request follows the instructions at [App Submission][submission].
- [X] I am using only the minimal set of permissions. *(If not, please explain each non-standard permission.)*
- [X] All assets referenced in the manifest are redistributable by any party.  If not, the unredistributable parts are using an extra-data source type.
- [ ] I am an upstream contributor to the project. If not, I contacted upstream developers about submitting their software to Flathub. **Link: https://github.com/unknown-horizons/unknown-horizons/issues/2775 (agreed by maintainer)**
- [X] I own the domain used in the application ID or the domain has a policy for delegating subdomains (e.g. GitHub, SourceForge).
Owned by Unknown Horizons themselves
- [X] Any additional patches or files have been submitted to the upstream projects concerned. *(If not, explain why.)*

[reqs]: https://github.com/flathub/flathub/wiki/App-Requirements
[maint]: https://github.com/flathub/flathub/wiki/App-Maintenance
[submission]: https://github.com/flathub/flathub/wiki/App-Submission

### CODE REVIEW

**Comment by LeandroStanger** [comment]
bot, build org.unknown_horizons.UnknownHorizons

File: org.unknown_horizons.UnknownHorizons.json [hunk 1]
```diff
@@ -0,0 +1,216 @@
...
+        "--device=dri",
+        "--socket=wayland",
+        "--socket=fallback-x11",
+        "--socket=pulseaudio",
+        "--share=network"
+    ],
+    "modules": [
+        {
+            "name": "fifechan",
+            "buildsystem": "cmake",
```
Comment (barthalion) [inline org.unknown_horizons.UnknownHorizons.json:20]:
```suggestion
            "buildsystem": "cmake-ninja",
```

File: org.unknown_horizons.UnknownHorizons.json [hunk 2]
```diff
@@ -0,0 +1,216 @@
...
+                {
+                    "type": "git",
+                    "url": "https://github.com/fifengine/fifechan.git",
+                    "tag": "0.1.5"
+                }
+            ]
+        },
+        {
+            "name": "fifengine",
+            "buildsystem": "cmake",
```
Comment (barthalion) [inline org.unknown_horizons.UnknownHorizons.json:32]:
```suggestion
            "buildsystem": "cmake-ninja",
```

**Review by barthalion — COMMENTED** [review]
It seems to miss an appdata file.

**Comment by tintou** [comment]
Thanks for the hint with cmake-ninja
It seems that the appdata was indeed not installed, I've now added it.

File: org.unknown_horizons.UnknownHorizons.json [hunk 3]
```diff
@@ -0,0 +1,219 @@
...
+    "app-id": "org.unknown_horizons.UnknownHorizons",
+    "runtime": "org.freedesktop.Platform",
+    "runtime-version": "20.08",
+    "sdk": "org.freedesktop.Sdk",
+    "command": "unknown-horizons",
+    "rename-desktop-file": "unknown-horizons.desktop",
+    "rename-appdata-file":  "unknown-horizons.appdata.xml",
+    "rename-icon": "unknown-horizons",
+    "finish-args": [
+        "--share=ipc",
```
Comment (bilelmoussaoui) [inline org.unknown_horizons.UnknownHorizons.json:11]:
```suggestion
```
you have that with fallback-x11 :)

**Comment by bilelmoussaoui** [comment]
bot, build org.unknown_horizons.UnknownHorizons

File: org.unknown_horizons.UnknownHorizons.json (hunk 3 above)
Comment (barthalion) [reply org.unknown_horizons.UnknownHorizons.json:11]: I looked at this after recent comment by @mwleeds asking if I'm really sure and – not really, I can't find lines that would imply it's true.

Comment (bilelmoussaoui) [reply org.unknown_horizons.UnknownHorizons.json:11]: Ah, my bad. I assumed it's after reading that somewhere on a review here :) thanks for the confirmation that it's not really a thing

**Comment by AsciiWolf** [comment]
You are missing version information in the AppStream metadata. This can be added as a patch, look [here](https://github.com/flathub/com.github.micahflee.torbrowser-launcher/blob/master/fix-AppData-file.patch) for example.

**Comment by tintou** [comment]
Indeed, I'll provide a patch upstream about this, a patch is now included in the mean time

**Comment by AsciiWolf** [comment]
Upstream patch for this is not needed and not a good idea if it won't be (auto)updated before new release.

**Comment by tintou** [comment]
I'm already using the master branch as I rely on several fixes for python 3.8 version that are not available in the latest release, my patches just got merged so I'll remove them from here :slightly_smiling_face:

**Comment by barthalion** [comment]
bot, build org.unknown_horizons.UnknownHorizons

**Pull request closed without merging**